Sources/MongoSwift/Cursor.swift

```diff
@@ -24,19 +24,26 @@ public class MongoCursor: Sequence, IteratorProtocol {
      * Close the cursor
      */
     public func close() {
+        self._client = nil
         guard let cursor = self._cursor else {
             return
         }
 
         mongoc_cursor_destroy(cursor)
         self._cursor = nil
-        self._client = nil
     }
 
     /**
      * Returns the next document in this cursor, or nil
      */
     public func next() -> Document? {
+        do {
+            _ = try unwrapCursor()
```

Comment:
while we should technically save and use this later on in the function rather than the direct accesses to `self._cursor`, I wrote it this way to avoid a big diff from wrapping everything in a `do/catch`since the function can't throw... the state of this function will be pretty temporary considering the changes I have now separated out to deal with cursor errors